src/main/java/com/avispl/symphony/dal/communicator/management/zoom/rooms/ZoomRoomsCommunicator.java

```
@@ -0,0 +1,395 @@
+        });
+    }
+
+    /**
+     * Parse properties parameters based on the provided start of the line,
+     * so the expected format of data provided here is:
+     *      Video Camera Line 1 Name: Camera Name
+     *      .....
+     *      Video Camera Line 2 Name: Camera Name
+     *
+     * @param response string from Zoom Rooms CLI
+     * @param propertyIdentifier supposed start of the string
+     * @return map of a format <Video Camera Line 1 Name: Camera Name, Video Camera Line 2 Name: Camera Name....>
+     */
+    private Map<String, String> parseZoomRoomsProperties(String response, String propertyIdentifier){
+        Map<String, String> parameters = new HashMap<>();
+        Arrays.asList(response.split("\n")).forEach(s -> {
+            if(s.startsWith(propertyIdentifier)){
+                String[] values = s.split(":");
+                parameters.put(values[0].replace("*s ", ""), values[1].replace("\r", ""));
+            }
+        });
+        return parameters;
+    }
+
+    /**
+     * Join meeting that is currently in progress or start a scheduled one.
+     * @param value - contains dialString of a format %meetingNumber%.%meetingPassword%@zoomcrc.com
+     *                password is optional.
+     * @return meetingId if operation is successful
+     */
+    private String joinStartMeeting(String value) throws Exception {
+        boolean hasPassword;
+        String meetingNumber;
+        String meetingPassword;
+
+        Pattern dialStringPattern = Pattern.compile("(\\d+)\\.?(\\d+)?(@zoomcrc\\.com)");
+        Matcher matcher = dialStringPattern.matcher(value);
+        if(!matcher.find() && !matcher.matches()){
+            // This might be the section to call out SIP/h323 devices it "invite" functionality fits our needs
+            if(logger.isErrorEnabled()) {
+                logger.error("Dial string does not match expected pattern meetingNumber.meetingPassword@zoomcrc.com. Skipping.");
+            }
+            return "";

```

Hmm I see that all over places. You cannot return "" or null for errors. Error means only one thing - exception thrown.

---

Agree, will make this explicit with exceptions thrown instead of just skipping it (basically)